src/Microsoft.Identity.Client/ClientAssertion.cs

```diff
@@ -46,5 +60,10 @@ public ClientAssertion(X509Certificate2 certificate, Dictionary<string, string>
         /// Gets the certificate
         /// </summary>
         public X509Certificate2 Certificate { get; set; }
+
+        /// <summary>
+        /// Gets the Signed Assertion
+        /// </summary>
+        public string SignedAssertion { get; set; }
```

Comment:
Prefer to have the SingedAssertion as a read-only property (i.e. remove the "set"). If there isn't an actual scenario where the user needs to change a property in the class, make it read-only. Immutable objects are much simpler to deal / code is simpler to read etc. 

Sadly the Certificate also has a setter, but we can't remove that for backwards compat reasons.

Reply:
I do not know if this should be read only or not. @jmprieur  will there ever be a scenario where the user needs to change the client assertion values after the CCA is constructed?

Reply:
We start with read-only, non-mutable options. Then we can change it if a scenario comes up. IMO, a new ConfidentialClientRequest object should be created if the client assertion needs to be changed.